app/src/main/java/org/schabi/newpipe/util/URLHandler.java

```diff
@@ -0,0 +1,124 @@
+package org.schabi.newpipe.util;
+
+import android.content.Context;
+
+import org.schabi.newpipe.extractor.NewPipe;
+import org.schabi.newpipe.extractor.StreamingService;
+import org.schabi.newpipe.extractor.exceptions.ExtractionException;
+import org.schabi.newpipe.extractor.exceptions.ParsingException;
+import org.schabi.newpipe.extractor.linkhandler.LinkHandlerFactory;
+import org.schabi.newpipe.extractor.stream.StreamInfo;
+import org.schabi.newpipe.player.playqueue.PlayQueue;
+import org.schabi.newpipe.player.playqueue.SinglePlayQueue;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers;
+import io.reactivex.rxjava3.core.Single;
+import io.reactivex.rxjava3.schedulers.Schedulers;
+
+public final class URLHandler {
```

Comment:
The name of this class is not meaningful enough. Something like DescriptionUrlHandler would be better (suggest a better one if you have one).

Reply:
I don't have a better for now, DescriptionUrlsHandler.

Reply:
This code is used for descriptions and comments, so I think URLsHandler is the best one.

Reply:
What about InternalUrlHandler? This would help distinguishing between the extractor's url handlers

Reply:
Done